src/Microsoft.AspNet.Mvc.TagHelpers/TagHelperOutputExtensions.cs

```diff
@@ -35,8 +35,16 @@ public static class TagHelperOutputExtensions
             {
                 // We look for the original attribute so we can restore the exact attribute name the user typed.
                 // Approach also ignores changes made to tagHelperOutput[attributeName].
-                var entry = context.AllAttributes.First(
+                var entry = context.AllAttributes.FirstOrDefault(
                     attribute => attribute.Key.Equals(attributeName, StringComparison.OrdinalIgnoreCase));
+
+                if (entry.Equals(default(KeyValuePair<string, object>)))
```

Comment:
> Perhaps, the `AllAttributes` dictionary should ignore case, which makes obtaining the value by key easier too.

`AllAttributes` is set up to ignore case.  See https://github.com/aspnet/Razor/blob/dev/src/Microsoft.AspNet.Razor.Runtime/TagHelpers/TagHelperExecutionContext.cs#L63

the code here is _not_ just looking up the correct value; it's trying to ensure the attribute is restored with the original attribute name i.e. correct case.  should remain as-is.

and the `entry.Equals()` check here is correct.  thanks for this change.